cabal-install/Distribution/Client/PackageHash.hs

```diff
@@ -319,3 +321,8 @@ hashFromTUF (Sec.Hash hashstr) =
         -> HashValue hash
       _ -> error "hashFromTUF: cannot decode base16 hash"
 
+-- | Hash a 'SourceRepo'. Excludes 'repoSubdir' field before
+-- hashing.
+hashSourceRepo :: SourceRepo -> HashValue
+hashSourceRepo sourceRepo =
+  hashValue $ Binary.encode (sourceRepo { repoSubdir = Nothing })
```

Comment:
So if I understand correctly, this is just to pick the location of the repository or is this to calculate the source hash for the content of the repo. In which case I don't think this belongs in this module which is about the package id hashes which are based on content hashes.